components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -120,16 +120,21 @@ public void addCORSOrigins(String applicationId, List<String> origins, String te
         List<Origin> originList = CORSConfigurationUtils.createOriginList(origins);
 
         // Check if the CORS origins are already present.
-        List<CORSOrigin> existingCORSOrigins = getCORSOriginDAO().getCORSOriginsByApplicationId(
-                applicationBasicInfo.getApplicationId(), tenantId);
-        for (Origin origin : originList) {
-            if (existingCORSOrigins.stream().map(CORSOrigin::getId).collect(Collectors.toList())
-                    .contains(origin.getValue())) {
-                // CORS origin is already registered for the application.
-                if (log.isDebugEnabled()) {
-                    log.debug(String.format(ERROR_CODE_ORIGIN_PRESENT.getMessage(), tenantDomain, origin));
+        if (applicationBasicInfo != null) {
```

Comment:
throw if applicationBasicInfo==null here . Otherwise there is a possible null pointer at line 142

Reply:
Handled in `getApplicationBasicInfo` private method.